src/main/java/org/myrobotlab/service/InMoov.java

```diff
@@ -1762,29 +1765,30 @@ public void loadGestures() {
    *          - the directory that contains the gesture python files.
    */
   public void loadGestures(String directory) {
-    // TODO: iterate over each of the python files in the directory
+
+    // iterate over each of the python files in the directory
     // and load them into the python interpreter.
+    String extension = "py";
+    Integer totalLoaded = 0;
+    Integer totalError = 0;
     File dir = makeGesturesDirectory(directory);
-    for (File f : dir.listFiles()) {
-      if (f.getName().toLowerCase().endsWith(".py")) {
-        log.info("Loading Gesture Python file {}", f.getAbsolutePath());
-        Python p = (Python) Runtime.getService("python");
-        String script = null;
-        try {
-          script = FileIO.toString(f.getAbsolutePath());
-        } catch (IOException e) {
-          log.warn("IO Error loading gesture file {} -- {}", f.getAbsolutePath(), e);
-          continue;
-        }
-        // evaluate the gestures scripts in a blocking way.
-        boolean result = p.exec(script, true, true);
-        if (!result) {
-          log.warn("Error while loading gesture file: {}", f.getAbsolutePath());
-        } else {
-          log.info("Successfully loaded gesture {}", f.getAbsolutePath());
+    if (dir.exists()) {
+      for (File f : dir.listFiles()) {
+        Boolean fileLoaded = Utils.loadFile(f.getAbsolutePath(), extension);
```

Comment:
This is very awkward to treat a boolean value as true, false, or null.  As i mentioned before,  I think you should test the file extension before you try to load the file here.  the logic would be a lot simpler.

Reply:
done